### PR TITLE
feat(dashboards): Improve loading states for widgets

### PR DESCRIPTION
**Pull request by dashed**
Use spinners. 

https://user-images.githubusercontent.com/139499/106068199-3265e480-60ce-11eb-9c52-9e6a173f6d65.mp4

### CODE REVIEW

**Review by wmak — APPROVED** [review]
LGTM ⏳

File: src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx [hunk 1]
```diff
@@ -487,3 +488,22 @@ class WidgetCardVisuals extends React.Component<WidgetCardVisualsProps> {
...
+
+const StyledTransparentLoadingMask = styled(props => (
+  <TransparentLoadingMask {...props} maskBackgroundColor="transparent" />
+))`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const LoadingScreen = ({loading}: {loading: boolean}) => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx:500]: Is this style of loader something we should be using with other charts as well?

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx:500]: I had borrowed this from Discover query cards.

It seems the pattern for pages where it has only 1 chart, the transparent mask is used instead; while letting the chart remain interactive.

File: src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx [hunk 2]
```diff
@@ -487,3 +488,22 @@ class WidgetCardVisuals extends React.Component<WidgetCardVisualsProps> {
...
+  if (!loading) {
+    return null;
+  }
+  return (
+    <StyledTransparentLoadingMask visible={loading}>
+      <LoadingIndicator mini />
```
Comment (k-fish) [inline src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx:506]: Oh I like the little loading indicator.

**Review by k-fish — APPROVED** [review]

File: src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx:500]: Hm... would this result in way too many spinners in different positions for other pages?

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**